### PR TITLE
Various Liquibase improvements

### LINKED ISSUE #7956: Slow startup with liquibase extension
**Describe the bug**
I added the liquibase extension to my project.
I have not written any big migrations yet, I just wanted to try and start with the extension present.
My startup time is immediatly around 2.5 seconds worse than without liquibase.

**Expected behavior**
No big impact from just having liquibase extension present.
It is not doing any migrations.
I would understand if it took some time to do migrations. But in this example no migrations are present.

**Actual behavior**
Startup time is 2.5 seconds slower with liquibase present

**To Reproduce**
1. Download the reproducer: 
[code-with-quarkus.zip](https://github.com/quarkusio/quarkus/files/4353511/code-with-quarkus.zip)

2. Start the dev mode.
3. It should say ".. started in 3s"
4. Now, add the "quarkus-liquibase" extension
5. Start devmode
6. It says ".. started in 6s"

**Configuration**
```properties
quarkus.datasource.url=jdbc:h2:tcp://localhost/mem:default
quarkus.datasource.driver=org.h2.Driver
quarkus.datasource.username=username-default
quarkus.datasource.min-size=3
quarkus.datasource.max-size=13
```

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
MSYS_NT-10.0 SOMEHOST 2.10.0(0.325/5/3) 2018-06-13 23:34 x86_64 Msys
 - Output of `java -version`: 
java version "1.8.0_241"
Java(TM) SE Runtime Environment (build 1.8.0_241-b07)
Java HotSpot(TM) 64-Bit Server VM (build 25.241-b07, mixed mode)
 - Quarkus version or git rev: 
1.3.0.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: C:\eclipse\tools\apache-maven\bin\..
Java version: 1.8.0_181, vendor: Oracle Corporation, runtime: C:\Program Files\Java\jdk1.8.0_181\jre
Default locale: de_DE, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

### PR DESCRIPTION
**Pull request by gsmet**
Should hopefully fix #7956

### CODE REVIEW

File: extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java [hunk 1]
```diff
@@ -54,26 +55,21 @@
...
     CapabilityBuildItem capability() {
         return new CapabilityBuildItem(Capabilities.LIQUIBASE);
     }
 
-    @BuildStep
-    void reflection(BuildProducer<ReflectiveClassBuildItem> reflective,
-            BuildProducer<NativeImageResourceBuildItem> resourceProducer,
-            BuildProducer<GeneratedResourceBuildItem> generatedResourceProducer,
-            BuildProducer<RuntimeInitializedClassBuildItem> runtimeInitialized) {
+    @BuildStep(onlyIf = NativeBuild.class, loadsApplicationClasses = true)
```
Comment (gsmet) [inline extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java]: This is the gist of it: only analyzing the changelog in the native image case as we don't need it in other cases.

File: extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java [hunk 2]
```diff
@@ -54,26 +55,21 @@
 
     private static final Logger LOGGER = Logger.getLogger(LiquibaseServiceLoader.class);
 
-    /**
-     * The change log parser factory singleton used for the liquibase service classpath scanner
-     */
-    private static final ChangeLogParserFactory CHANGE_LOG_PARSER_FACTORY = ChangeLogParserFactory.getInstance();
```
Comment (gsmet) [inline extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java]: This is where all the time is passed: it has a very flawed and slow scanning process.

We now only create that instance when building a native image.

Comment (SteveDonie) [reply extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java]: Hi - I am the Liquibase Community Manager. Wanted to let y'all know that this is something that should be improved in our next major release, 4.0.0. Keep an eye on the blog for more info. 4.0 Beta 1 should be available in the next month.

Comment (gsmet) [reply extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java]: @SteveDonie ah cool! Feel free to ping us if you want to discuss things.

**Review by machi1990 — APPROVED** [review]
LGTM and @Postremus confirmed that it fixed the perf issue https://github.com/quarkusio/quarkus/issues/7956#issuecomment-601826050.

**Pull request merged**